cpp/src/Slice/Parser.cpp

```diff
@@ -781,7 +783,14 @@ Slice::Contained::parseDocComment(function<string(string, string)> linkFormatter
         string lineText;
         if (parseCommentLine(l, paramTag, true, name, lineText))
         {
-            if (!lineText.empty())
+            if (!isOperation)
+            {
+                // If '@param' was put on anything other than an operation, ignore it and issue a warning.
+                string msg = "the '" + paramTag + "' tag is only valid on operations";
+                unit()->warning(file(), line(), InvalidComment, msg);
+                state = StateUnknown;
+            }
+            else if (!lineText.empty())
```

Comment:
I think it would make sense to emit a warning if lineText is empty. 

Do we validate that the `@param` names, match the operation param names?